packages/flutter/lib/src/widgets/fade_in_image.dart

```diff
@@ -562,7 +562,10 @@ class _AnimatedFadeOutFadeInState extends ImplicitlyAnimatedWidgetState<_Animate
 
   @override
   Widget build(BuildContext context) {
-    if (widget.wasSynchronouslyLoaded || _placeholderOpacityAnimation!.isCompleted) {
+    final bool placeholderIsNotAnimating =
```

Comment:
just inline this in the if condition block to avoid additional calculation if widget.wasSynchronouslyLoaded is true

Reply:
was attempting to create an informative variable, for better understanding. will inline it.